### ISSUE TITLE
Getting NAN after few iterations

### ISSUE THREAD

**Issue by dhirajsuvarna** [fxia22/pointnet.pytorch#58]
Hi, 
I am using PointNet on Pytorch to train my custom dataset.
However, after training around 10-20 batches of examples, the loss goes into NAN.

I highly doubt that it is a exploding gradient issue. (still trying to diagnose the problem)

Posting this problem here in expectation of any help or direction that you may provide.

Things I have tried is - 
1. Reducing the Learning Rate to 0.0001
2. setting drop_last=True in DataLoader
3. reducing the batch size from 32 to 10

**Comment by fxia22** [comment]
Can you also make sure your data doesn't have nan?

**Comment by dhirajsuvarna** [comment]
I dont think the dataset contains any NAN.
The dataset contains points, which are generated from 3d CAD Models.

Anyways, will write a script to look at this and update you.

**Comment by onlinehuazai** [comment]
> @fxia22,
> 
> You were right, there were NAN's in the input to the network.
> These NaN's were generated after the Zero Mean and Normalization step for one model only.
> 
> For time being I have removed that particular model from the dataset and was able to train.
> 
> Thank You for your help.

how to solve this problem?
